applications/GeoMechanicsApplication/tests/cpp_tests/custom_elements/test_integration_schemes.cpp

```diff
@@ -0,0 +1,130 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Anne van de Graaf
+//
+
+#include "custom_elements/lobatto_integration_scheme.h"
+#include "includes/expect.h"
+#include "tests/cpp_tests/geo_mechanics_fast_suite.h"
+
+using namespace Kratos;
+
+namespace
+{
+
+std::unique_ptr<IntegrationScheme> MakeLobattoIntegrationScheme(std::size_t NumberOfPoints)
+{
+    return std::make_unique<LobattoIntegrationScheme>(NumberOfPoints);
+}
+
+double SumOfWeights(const Geo::IntegrationPointVectorType& rIntegrationPoints)
+{
+    auto weights = std::vector<double>{};
+    std::transform(rIntegrationPoints.begin(), rIntegrationPoints.end(), std::back_inserter(weights),
+                   [](const auto& rIntegrationPoint) { return rIntegrationPoint.Weight(); });
+    return std::accumulate(weights.cbegin(), weights.cend(), 0.0);
+}
+
+void ExpectLocalCoordinatesIncludeRangeBounds(const Geo::IntegrationPointVectorType& rIntegrationPoints)
+{
+    constexpr auto tolerance = 1.0e-6;
+
+    auto contains_lower_bound_of_xi = [tolerance](const auto& rPoint) {
+        return std::abs(rPoint[0] + 1.0) <= tolerance;
+    };
+    KRATOS_EXPECT_TRUE(std::any_of(rIntegrationPoints.begin(), rIntegrationPoints.end(), contains_lower_bound_of_xi))
+
+    auto contains_upper_bound_of_xi = [tolerance](const auto& rPoint) {
+        return std::abs(rPoint[0] - 1.0) <= tolerance;
+    };
+    KRATOS_EXPECT_TRUE(std::any_of(rIntegrationPoints.begin(), rIntegrationPoints.end(), contains_upper_bound_of_xi))
+}
+
+void ExpectLocalCoordinatesAreInRange(const Geo::IntegrationPointVectorType& rIntegrationPoints)
+{
+    constexpr auto tolerance = 1.0e-6;
+
+    auto xi_is_in_range = [tolerance](const auto& rPoint) {
+        return std::abs(rPoint[0]) - 1.0 <= tolerance;
+    };
+    KRATOS_EXPECT_TRUE(std::all_of(rIntegrationPoints.begin(), rIntegrationPoints.end(), xi_is_in_range))
+
+    auto non_xi_coordinates_must_be_near_zero = [tolerance](const auto& rPoint) {
+        return (std::abs(rPoint[1]) <= tolerance) && (std::abs(rPoint[2]) <= tolerance);
+    };
```

Comment:
Probably me, but I had to read these lines twice to understand the name. Is it an idea to add a comment to clarify that since this is a 1D local space, only the first component is filled?

Reply:
Added a comment.